### PR TITLE
Hydrogen Port 2: Electric Boogaloo

### PR DESCRIPTION
**Pull request by Hockaa**
This is a redo of #7456, so I will copy (some of) the description from that PR:

Ports Hydrogen gas from Baystation - a flammable gas typically used as fuel. For now, only useful as an alternative coolant for the Supermatter. The nitrogen canisters in Engineering Hard Storage are replaced with four Hydrogen canisters.

I also took the liberty of removing the NO2 define and changing the atmos_suitable var to TRUE/FALSE instead of 1/0

### CODE REVIEW

**Review by furrycactus — APPROVED** [review]

File: code/game/machinery/atmoalter/canister.dm [hunk 1]
```diff
@@ -475,6 +492,11 @@ update_flag
 	src.air_contents.adjust_gas(GAS_PHORON, MolesForPressure())
 	src.update_icon()
 
+/obj/machinery/portable_atmospherics/canister/hydrogen/engine_setup/Initialize()
```
Comment (mikomyazaki) [inline code/game/machinery/atmoalter/canister.dm:495]: This is identical to its parent, which you also define.

Comment (Hockaa) [reply code/game/machinery/atmoalter/canister.dm:495]: Didn't actually think about that when I was shoving the code over - It looks like all of the engine_setup variants are literally the same as their parents, the only difference is that they are used in the setup_supermatter.dm file, and nowhere else? I'll just make the setup file use the parent canisters and delete the engine_setup variants.

File: code/ZAS/Turf.dm [hunk 2]
```diff
@@ -260,6 +260,7 @@
 		GM.gas[GAS_CO2] = (carbon_dioxide/sum)*amount
 		GM.gas[GAS_NITROGEN] = (nitrogen/sum)*amount
 		GM.gas[GAS_PHORON] = (phoron/sum)*amount
+		GM.gas[GAS_HYDROGEN] = (phoron/sum)*amount
```
Comment (furrycactus) [inline code/ZAS/Turf.dm:263]: Noticed this on another quick scan. You've got phoron/sum on Hydrogen.

**Review by mikomyazaki — APPROVED** [review]

**Review by fernerr — DISMISSED** [review]
hydrogen
The supermatter control console 
(/obj/machinery/computer/general_air_control/supermatter_core)
does not list hydrogen in its gas composition, might be related to /obj/machinery/air_sensor not picking up hydrogen readings and/or the console not handling the data correctly.

File: code/game/machinery/alarm.dm [hunk 3]
```diff
@@ -542,10 +548,11 @@
...
 						"filters"	= list()
 					)
-				scrubbers[scrubbers.len]["filters"] += list(list("name" = GAS_OXYGEN,			"command" = "o2_scrub",	"val" = info["filter_o2"]))
+				scrubbers[scrubbers.len]["filters"] += list(list("name" = GAS_OXYGEN,		"command" = "o2_scrub",	"val" = info["filter_o2"]))
 				scrubbers[scrubbers.len]["filters"] += list(list("name" = GAS_NITROGEN,		"command" = "n2_scrub",	"val" = info["filter_n2"]))
-				scrubbers[scrubbers.len]["filters"] += list(list("name" = GAS_CO2, "command" = "co2_scrub","val" = info["filter_co2"]))
-				scrubbers[scrubbers.len]["filters"] += list(list("name" = "Phoron"	, 		"command" = "tox_scrub","val" = info["filter_phoron"]))
+				scrubbers[scrubbers.len]["filters"] += list(list("name" = GAS_CO2, 			"command" = "co2_scrub","val" = info["filter_co2"]))
+				scrubbers[scrubbers.len]["filters"] += list(list("name" = GAS_PHORON, 		"command" = "tox_scrub","val" = info["filter_phoron"]))
+				scrubbers[scrubbers.len]["filters"] += list(list("name" = GAS_HYDROGEN,		"command" = "h2_scrub","val" = info["filter_h2"]))
```
Comment (NonQueueingMatt) [inline code/game/machinery/alarm.dm:555]: The GAS_X defines are lowercased so we don't want to use them here.

**Pull request merged**